src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Split.cs

```diff
@@ -87,7 +87,36 @@ private static string[] Split(Regex regex, string input, int count, int startat)
             count--;
             (List<string> results, int prevat, string input, int count) state = (results: new List<string>(), prevat: 0, input, count);
 
-            if (!regex.RightToLeft)
+            if (regex._srm is not null)
+            {
+                Match match = regex.Match(input, startat, input.Length - startat);
+                if (!match.Success)
+                {
+                    // If there is no match then return the input as the only part in the split
+                    return new[] { input };
+                }
+
+                while (match.Success)
+                {
+                    // Add the substring before the found match, this may be empty
+                    state.results.Add(state.input.Substring(state.prevat, match.Index - state.prevat));
+                    state.prevat = match.Index + match.Length;
+
+                    if (--state.count == 0)
+                    {
+                        // If the maximum number of allowed mathes is reached break the splitting
+                        // If the initial value of count is 0 or negative this value just keeps decreasing further
+                        break;
+                    }
+
+                    // Omit the matched substring itself and find the next match
```

Comment:
What does "Omit the matched substring itself" mean here?

Reply:
Well, the match itself is excluded from the output, but that I guess is the whole point of Split, so perhaps not such a helpful comment.

Reply:
I will remove this comment.